comit/src/network/protocols/orderbook/take_order.rs

```diff
@@ -0,0 +1,116 @@
+use crate::network::OrderId;
+use futures::{prelude::*, AsyncWriteExt};
+use libp2p::{
+    core::upgrade,
+    request_response::{ProtocolName, RequestResponseCodec},
+};
+use serde::Deserialize;
+use std::io;
+use tracing::debug;
+
+#[derive(Clone)]
+pub struct TakeOrderProtocol;
+
+impl ProtocolName for TakeOrderProtocol {
+    fn protocol_name(&self) -> &[u8] {
+        b"/comit/orderbook/take/1.0.0"
+    }
+}
+
+#[derive(Clone, Default)]
+pub struct TakeOrderCodec;
+
+/// The different responses we can send back as part of an announcement.
+///
+/// For now, this only includes a generic error variant in addition to the
+/// confirmation because we simply close the connection in case of an error.
+#[derive(Debug, Deserialize)]
+pub enum Response {
+    Confirmation,
```

Comment:
Confirmation should include the `SharedSwapId`.

Reply:
Stale, I should have read this more carefully 4 days ago ...